suite-native/device-manager/src/components/DeviceItem/WalletDetailDeviceItemContent.tsx

```diff
@@ -0,0 +1,67 @@
+import { useSelector } from 'react-redux';
+
+import { HStack, Text, Box } from '@suite-native/atoms';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { DeviceRootState, selectDeviceByState } from '@suite-common/wallet-core';
+
+import { ConnectionDot } from './ConnectionDot';
+import { SwitchContentProps } from '../../types';
+
+const headerStyle = prepareNativeStyle(utils => ({
+    flexShrink: 1,
+    paddingRight: utils.spacings.small,
+    alignItems: 'center',
+    gap: utils.spacings.small,
+}));
+const headerTextStyle = prepareNativeStyle(() => ({
+    flexShrink: 1,
+}));
+
+export const WalletDetailDeviceItemContent = ({
+    deviceState,
+    headerTextVariant,
+    isPortfolioLabelDisplayed,
+    header,
+    subHeader,
+    isPortfolioTrackerDevice,
+}: SwitchContentProps) => {
+    const { applyStyle } = useNativeStyles();
+    const device = useSelector((state: DeviceRootState) =>
+        deviceState !== undefined ? selectDeviceByState(state, deviceState) : undefined,
+    );
+
+    return (
+        <>
+            <HStack style={applyStyle(headerStyle)}>
+                <Text
+                    variant={headerTextVariant}
+                    ellipsizeMode="tail"
+                    numberOfLines={1}
+                    style={applyStyle(headerTextStyle)}
+                >
+                    {header}
+                </Text>
+                {!isPortfolioTrackerDevice && device && (
+                    <ConnectionDot isConnected={!!device?.connected} />
```

Comment:
selectIsDeviceConntected, also handle optional device above render, same for the optional few lines below